app/models/TraderProfile.scala

```diff
@@ -41,31 +40,29 @@ object TraderProfile {
       answers.getOptionalPageValue(answers, HasNiphlPage, NiphlNumberPage)
     ).parMapN(TraderProfile.apply)
 
-  def buildNirms(
+  private def getOptionallyRemovedPage(
```

Comment:
this file had tons of duplicated code and different ways of doing each bit which was causing issues when i was trying to test the different scenarios for the banners

so i stripped it back and picked the best bits from each function so now there is just one function (this one) that you need to understand